### ISSUE TITLE
[I] QownNotes Doesn't Ask For Password To Decrypt A Note, Resulting in Gibberish

### ISSUE THREAD

**Issue by DanteIgashu** [pbek/QOwnNotes#2571]
#### Expected behaviour

1. QownNotes asks for password to decrypt note
2. QownNotes accepts password and decrypts note
3. QownNotes show the actual text

#### Actual behaviour

1. QownNotes doesn't ask for password to decrypt note
2. QownNotes "decrypts" note anyway
3. QownNotes shows gibberish instead of the actual text in the note

#### Steps to reproduce

1. Select note
2. Select "decrypt"

#### Relevant log output in the Log panel

<details><summary>Expand</summary>

```
[15:11:53] [warning] Wayland does not support QWindow::requestActivate()
[15:12:03] [warning] Wayland does not support QWindow::requestActivate()

That's it. No errors.
(Is there a way to decrypt the note outside of QownNotes, or make QownNotes actually decrypt it?)

```
</details>

#### Information about the application, settings and environment

<details><summary>Expand</summary>

QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `Sun Aug 7 15:12:43 2022`
**Version**: `22.8.0`
**Build date**: `Aug  2 2022`
**Build number**: `977`
**Platform**: `linux`
**Operating System**: `KDE Flatpak runtime`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `TuxFamily`
**Qt Version (build)**: `5.15.5`
**Qt Version (runtime)**: `5.15.5`
**Portable mode**: `no`
**Settings path / key**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/config/PBE/QOwnNotes.conf`
**Application database path**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/data/PBE/QOwnNotes/QOwnNotes.sqlite`
**Application arguments**: `QOwnNotes`
**Qt Debug**: `0`
**Locale (system)**: `en_US`
**Locale (interface)**: *empty*
**Primary screen resolution**: `1920x1080`
**Icon theme**: `breeze-dark-qownnotes`
**Notes in current note folder**: `3`
**Calendar items**: `0`
**Enabled scripts**: `0`

## Server Info

**serverUrl**: *empty*
**appIsValid**: `no`
**notesPathExists**: *empty*
**connectionErrorMessage**: *empty*

## Spellchecking

**Enabled**: `true`
**Selected language**: `auto`
**Language codes**: `en_AU, en_BS, en_BZ, en_CA, en_GB, en_GH, en_IE, en_IN, en_JM, en_MW, en_NA, en_NZ, en_PH, en_TT, en_US, en_ZA, en_ZW`
**Language names**: `Australian English (Australia), English (Bahamas), English (Belize), Canadian English (Canada), British English (United Kingdom), English (Ghana), English (Ireland), English (India), English (Jamaica), English (Malawi), English (Namibia), English (New Zealand), English (Philippines), English (Trinidad & Tobago), American English (United States), English (South Africa), English (Zimbabwe)`
**Application dictionaries path**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/data/PBE/QOwnNotes/dicts`

## Note folders

**currentNoteFolderId**: `1`

### Note folder `default`

**id**: `1`
**isCurrent**: `yes`
**activeTagId**: `0`
**localPath**: `/home/pc_user/.local/share/Notes`
**remotePath**: `Notes`
**cloudConnectionId**: `1`
**isShowSubfolders**: `no`
**isUseGit**: `no`
**allowDifferentNoteFileName**: `no`
**activeNoteSubFolder name**: *empty*
**database file**: `/home/pc_user/.local/share/Notes/notes.sqlite`

### Cloud connection ``

**id**: `1`
**isCurrent**: `yes`
**serverUrl**: *empty*
**username**: *empty*
**accountId**: *empty*

## Settings

**ActiveNoteHistoryItem** (NoteHistoryItem): `<binary data>`
**Debug/fakeOldVersionNumber** (QString): `false`
**Debug/fileLogging** (QString): `false`
**Editor/CurrentSchemaKey** (QString): `EditorColorSchema-cdbf28fc-1ddc-4d13-bb21-6a4043316a2f`
**Editor/autoBracketClosing** (QString): `false`
**Editor/autoBracketRemoval** (QString): `true`
**Editor/disableCursorBlinking** (QString): `false`
**Editor/editorWidthInDFMOnly** (QString): `true`
**Editor/highlightCurrentLine** (QString): `true`
**Editor/indentSize** (QString): `4`
**Editor/removeTrailingSpaces** (QString): `false`
**Editor/showLineNumbers** (QString): `true`
**Editor/useTabIndent** (QString): `false`
**Editor/vimMode** (QString): `false`
**ImageDialog/disableCopying** (QString): `false`
**ImageDialog/geometry** (QByteArray): `<binary data>`
**IssueAssistantDialog/geometry** (QByteArray): `<binary data>`
**LastUpdateCheck** (QDateTime): `2022-08-07T15:06:37.440`
**LogWidget/criticalLog** (bool): `true`
**LogWidget/debugLog** (bool): `true`
**LogWidget/fatalLog** (bool): `true`
**LogWidget/infoLog** (bool): `true`
**LogWidget/scriptingLog** (bool): `true`
**LogWidget/statusLog** (bool): `true`
**LogWidget/warningLog** (bool): `true`
**MainWindow/geometry** (QByteArray): `<binary data>`
**MainWindow/mainToolBar.iconSize** (QString): `22`
**MainWindow/menuBarGeometry** (QByteArray): `<binary data>`
**MainWindow/noteTextEdit.code.font** (QString): `Unispace,8,-1,5,75,0,0,0,0,0,Bold`
**MainWindow/noteTextEdit.font** (QString): `Unispace,8,-1,5,75,0,0,0,0,0,Bold`
**MainWindow/noteTextView.code.font** (QString): `Unispace,8,-1,5,75,0,0,0,0,0,Bold`
**MainWindow/noteTextView.font** (QString): `Unispace,8,-1,5,75,0,0,0,0,0,Bold`
**MainWindow/noteTextView.ignoreCodeFontSize** (QString): `true`
**MainWindow/noteTextView.rtl** (QString): `false`
**MainWindow/noteTextView.underline** (QString): `true`
**MainWindow/noteTextView.useEditorStyles** (QString): `true`
**MainWindow/noteTextView.useInternalExportStyling** (QString): `true`
**MainWindow/showNotesFromAllNoteSubFolders** (QString): `true`
**NoteDialog/geometry** (QByteArray): `<binary data>`
**NoteFolder-1/NoteTabNameList** (QString): `My Notes`
**NoteFolder-1/NoteTabStickinessList**: *empty*
**NoteFolder-1/NoteTabSubFolderPathDataList** (QString): *empty*
**NoteHistory-1** (QVariantList): `<variant list with 3 item(s)>`
**NoteHistoryCurrentIndex-1** (QString): `2`
**PasswordDialog/geometry** (QByteArray): `<binary data>`
**PiwikClientId** (QString): `<hidden>`
**ScriptRepositoryDialog/mainSplitterState** (QByteArray): `<binary data>`
**SearchEngineId** (QString): `2`
**SettingsDialog/geometry** (QByteArray): `<binary data>`
**SettingsDialog/mainSplitterState** (QByteArray): `<binary data>`
**ShowSystemTray** (QString): `true`
**StartHidden** (QString): `true`
**UpdateDialog/geometry** (QByteArray): `<binary data>`
**WelcomeDialog/geometry** (QByteArray): `<binary data>`
**acceptAllExternalModifications** (QString): `false`
**allowNoteEditing** (bool): `true`
**allowOnlyOneAppInstance** (QString): `true`
**appMetrics/disableAppHeartbeat** (QString): `false`
**appMetrics/disableTracking** (QString): `false`
**appMetrics/notificationShown** (QString): `true`
**automaticNoteFolderDatabaseClosing** (QString): `false`
**checkSpelling** (bool): `true`
**closeTodoListAfterSave** (QString): `false`
**cryptoKey** (QString): `<hidden>`
**currentNoteFolderId** (QString): `1`
**currentWorkspace** (QString): `initial`
**cursorWidth** (QString): `1`
**darkMode** (QString): `true`
**darkModeColors** (QString): `true`
**darkModeIconTheme** (QString): `true`
**darkModeTrayIcon** (QString): `true`
**defaultNoteFileExtension** (QString): `md`
**demoNotesCreated** (QString): `true`
**disableAutomaticUpdateDialog** (QString): `true`
**disableSavedSearchesAutoCompletion** (QString): `false`
**dockWasInitializedOnce** (QString): `true`
**enableNoteTree** (QString): `false`
**enableSocketServer** (QString): `false`
**enableWebAppSupport** (QString): `false`
**externalEditorPath** (QString): *empty*
**fullyHighlightedBlockquotes** (QString): `false`
**gitCommitInterval** (QString): `30`
**gitExecutablePath** (QString): *empty*
**gitLogCommand** (QString): *empty*
**guiFirstRunInit** (bool): `true`
**ignoreAllExternalModifications** (QString): `false`
**ignoreAllExternalNoteFolderChanges** (QString): `false`
**ignoreNoteSubFolders** (QString): `^\.`
**imageScaleDown** (QString): `false`
**imageScaleDownMaximumHeight** (QString): `1024`
**imageScaleDownMaximumWidth** (QString): `1024`
**initialLayoutIdentifier** (QString): `minimal`
**insertTimeFormat** (QString): `yyyy.MM.dd dddd HH:mm:ss`
**interfaceFontSize** (QString): `11`
**interfaceLanguage** (QString): *empty*
**internalIconTheme** (QString): `false`
**itemHeight** (QString): `20`
**legacyLinking** (QString): `false`
**localTrash/autoCleanupDays** (QString): `30`
**localTrash/autoCleanupEnabled** (QString): `true`
**localTrash/supportEnabled** (QString): `true`
**markdownHighlightingEnabled** (QString): `true`
**navigationPanelAutoSelect** (QString): `true`
**navigationPanelHideSearch** (QString): `false`
**networking/ignoreSSLErrors** (QString): `true`
**networking/proxyType** (QString): `2`
**newNoteAskHeadline** (QString): `false`
**noteEditIsCentralWidget** (QString): `true`
**noteFileExtensionList** (QStringList): `md, txt`
**noteListPreview** (QString): `false`
**noteSaveIntervalTime** (QString): `10`
**noteSubfoldersPanelDisplayAsFullTree** (QString): `true`
**noteSubfoldersPanelHideSearch** (QString): `false`
**noteSubfoldersPanelOrder** (QString): `0`
**noteSubfoldersPanelShowFullPath** (QString): `false`
**noteSubfoldersPanelShowNotesRecursively** (QString): `false`
**noteSubfoldersPanelShowRootFolderName** (QString): `true`
**noteSubfoldersPanelSort** (QString): `0`
**notesPanelOrder** (QString): `0`
**notesPanelSort** (QString): `1`
**notesPath** (QString): `/home/pc_user/.local/share/Notes`
**notifyAllExternalModifications** (QString): `false`
**overrideInterfaceFontSize** (QString): `false`
**ownCloud/supportEnabled** (QString): `false`
**ownCloud/todoCalendarBackend** (QString): `3`
**ownCloud/todoCalendarCalDAVPassword** (QString): `<hidden>`
**ownCloud/todoCalendarCalDAVServerUrl** (QString): *empty*
**ownCloud/todoCalendarCalDAVUsername** (QString): *empty*
**ownCloud/todoCalendarCloudConnectionId** (QString): `0`
**ownCloud/todoCalendarDisplayNameList**: *empty*
**ownCloud/todoCalendarEnabledList**: *empty*
**ownCloud/todoCalendarEnabledUrlList**: *empty*
**ownCloud/todoCalendarUrlList**: *empty*
**restoreCursorPosition** (QString): `true`
**restoreLastNoteAtStartup** (QString): `true`
**restoreNoteTabs** (QString): `true`
**savedSearches/noteFolder-1** (QStringList): `Note 2022-02-18T16.16.10, My Notes`
**showMatches** (QString): `true`
**showMenuBar** (bool): `true`
**showStatusBar** (bool): `true`
**spellCheckLanguage** (QString): `auto`
**systemIconTheme** (QString): `false`
**taggingShowNotesRecursively** (QString): `false`
**tagsPanelHideNoteCount** (QString): `false`
**tagsPanelHideSearch** (QString): `false`
**tagsPanelOrder** (QString): `0`
**tagsPanelSort** (QString): `0`
**todoCalendarSupport** (QString): `true`
**toolbar/1/items** (QStringList): `action_New_note, action_Find_note, action_Remove_note, action_Open_note_in_external_editor, actionShow_local_trash, actionAllow_note_editing, , action_Back_in_note_history, action_Forward_in_note_history, , action_Find_text_in_note, actionReplace_in_current_note, , actionShow_versions, actionShow_trash, actionShare_note, , actionShow_Todo_List`
**toolbar/1/name** (QString): `mainToolBar`
**toolbar/1/title** (QString): `main toolbar`
**toolbar/2/items** (QStringList): `actionFormat_text_bold, actionFormat_text_italic, actionStrike_out_text, actionInsert_code_block, actionInsert_block_quote`
**toolbar/2/name** (QString): `formattingToolbar`
**toolbar/2/title** (QString): `formatting toolbar`
**toolbar/3/items** (QStringList): `actionInsert_text_link, actionInsert_image, actionInsert_current_time`
**toolbar/3/name** (QString): `insertingToolbar`
**toolbar/3/title** (QString): `inserting toolbar`
**toolbar/4/items** (QStringList): `action_Encrypt_note, actionEdit_encrypted_note, actionDecrypt_note`
**toolbar/4/name** (QString): `encryptionToolbar`
**toolbar/4/title** (QString): `encryption toolbar`
**toolbar/5/items** (QStringList): `actionWorkspaceComboBox, actionStore_as_new_workspace, actionRemove_current_workspace, actionRename_current_workspace, actionSwitch_to_previous_workspace, actionUnlock_panels, , actionToggle_distraction_free_mode, action_Increase_note_text_size, action_Decrease_note_text_size, action_Reset_note_text_size`
**toolbar/5/name** (QString): `windowToolbar`
**toolbar/5/title** (QString): `window toolbar`
**toolbar/6/items** (QString): `action_Quit`
**toolbar/6/name** (QString): `quitToolbar`
**toolbar/6/title** (QString): `quit toolbar`
**toolbar/size** (QString): `6`
**useNoteFolderButtons** (QString): `false`
**useUNIXNewline** (QString): `false`
**webAppClientService/serverUrl** (QString): `wss://app.qownnotes.org`
**webAppClientService/token** (QString): `<hidden>`
**webSocketServerService/bookmarksNoteName** (QString): `Bookmarks`
**webSocketServerService/bookmarksTag** (QString): `bookmarks`
**webSocketServerService/commandSnippetsNoteName** (QString): `Commands`
**webSocketServerService/commandSnippetsTag** (QString): `commands`
**webSocketServerService/port** (QString): `22222`
**workspace-initial/name** (QString): `Minimal`
**workspace-initial/noteSubFolderDockWidgetVisible** (QString): `true`
**workspace-initial/windowState** (QByteArray): `<binary data>`
**workspaces** (QString): `initial`

## System environment

**XDG_RUNTIME_DIR**: `/run/user/1000`
**container**: `flatpak`
**AT_SPI_BUS_ADDRESS**: `unix:path=/run/flatpak/at-spi-bus`
**BROWSER**: `firefox`
**DBUS_SESSION_BUS_ADDRESS**: `unix:path=/run/flatpak/bus`
**DESKTOP_SESSION**: `plasmawayland`
**DISPLAY**: `:99.0`
**DOTNET_BUNDLE_EXTRACT_BASE_DIR**: `/home/pc_user/.cache/dotnet_bundle_extract`
**DOTNET_ROOT**: `/usr/share/dotnet`
**EDITOR**: `nano`
**FLATPAK_ID**: `org.qownnotes.QOwnNotes`
**FLATPAK_SANDBOX_DIR**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/sandbox`
**GDK_DPI_SCALE**: `1`
**GDK_SCALE**: `1`
**GI_TYPELIB_PATH**: `/app/lib/girepository-1.0`
**GST_PLUGIN_PATH**: `/app/lib/gstreamer-1.0:/usr/lib/extensions/gstreamer-1.0:/usr/lib/x86_64-linux-gnu/gstreamer-1.0`
**GTK2_RC_FILES**: `/etc/gtk-2.0/gtkrc:/home/pc_user/.gtkrc-2.0:/home/pc_user/.config/gtkrc-2.0`
**GTK_RC_FILES**: `/etc/gtk/gtkrc:/home/pc_user/.gtkrc:/home/pc_user/.config/gtkrc`
**HOME**: `/home/pc_user`
**INVOCATION_ID**: `ecadd1578e2149d8a8e02133a1788bb5`
**JOURNAL_STREAM**: `8:20195`
**KDE_APPLICATIONS_AS_SCOPE**: `1`
**KDE_FORK_SLAVES**: `1`
**KDE_FULL_SESSION**: `true`
**KDE_SESSION_UID**: `1000`
**KDE_SESSION_VERSION**: `5`
**LANG**: `en_US.UTF-8`
**LC_ADDRESS**: `en_US.UTF-8`
**LC_IDENTIFICATION**: `en_US.UTF-8`
**LC_MEASUREMENT**: `es_MX.UTF-8`
**LC_MONETARY**: `es_MX.UTF-8`
**LC_NAME**: `en_US.UTF-8`
**LC_NUMERIC**: `en_US.UTF-8`
**LC_PAPER**: `en_US.UTF-8`
**LC_TELEPHONE**: `en_US.UTF-8`
**LC_TIME**: `en_GB.UTF-8`
**LD_LIBRARY_PATH**: *empty*
**LOGNAME**: `pc_user`
**MAIL**: `/var/spool/mail/pc_user`
**MANAGERPID**: `948`
**MOTD_SHOWN**: `pam`
**PAM_KWALLET5_LOGIN**: `/run/user/1000/kwallet5.socket`
**PATH**: `/app/bin:/usr/bin`
**PLASMA_USE_QT_SCALING**: `1`
**PS1**: `[📦 $FLATPAK_ID \W]\$ `
**PWD**: `/home/pc_user`
**QML2_IMPORT_PATH**: `/app/lib64/qml:/app/lib/qml`
**QT_AUTO_SCREEN_SCALE_FACTOR**: `0`
**QT_PLUGIN_PATH**: `/app/lib64/plugins:/app/lib/plugins:/usr/share/runtime/lib/plugins`
**QT_QPA_PLATFORMTHEME**: `qt5ct`
**QT_WAYLAND_FORCE_DPI**: `96`
**SESSION_MANAGER**: `local/PC:@/tmp/.ICE-unix/1121,unix/PC:/tmp/.ICE-unix/1121`
**SHELL**: `/bin/sh`
**SHLVL**: `0`
**SYSTEMD_EXEC_PID**: `1136`
**USER**: `pc_user`
**WAYLAND_DISPLAY**: `wayland-0`
**XAUTHORITY**: `/run/flatpak/Xauthority`
**XCURSOR_SIZE**: `24`
**XCURSOR_THEME**: `oxy-red-argentina`
**XDG_ACTIVATION_TOKEN**: `kwin-37`
**XDG_CACHE_HOME**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/cache`
**XDG_CONFIG_DIRS**: `/app/etc/xdg:/etc/xdg`
**XDG_CONFIG_HOME**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/config`
**XDG_CURRENT_DESKTOP**: `KDE`
**XDG_DATA_DIRS**: `/app/share:/usr/share:/usr/share/runtime/share:/run/host/user-share:/run/host/share`
**XDG_DATA_HOME**: `/home/pc_user/.var/app/org.qownnotes.QOwnNotes/data`
**XDG_SEAT**: `seat0`
**XDG_SEAT_PATH**: `/org/freedesktop/DisplayManager/Seat0`
**XDG_SESSION_CLASS**: `user`
**XDG_SESSION_DESKTOP**: `KDE`
**XDG_SESSION_ID**: `2`
**XDG_SESSION_PATH**: `/org/freedesktop/DisplayManager/Session1`
**XDG_SESSION_TYPE**: `wayland`
**XDG_VTNR**: `2`
**XKB_DEFAULT_LAYOUT**: `us`
**XKB_DEFAULT_VARIANT**: `altgr-intl`
**__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS**: `/etc/egl/egl_external_platform.d:/usr/lib/x86_64-linux-gnu/GL/egl/egl_external_platform.d:/usr/share/egl/egl_external_platform.d`
</details>

**Comment by pbek** [comment]
I'm sorry, I cannot reproduce your issue. I just encrypted and decrypted a note without fault.
(Btw. normally you want to edit or view a encrypted note, not actually decrypt it.)

Does that also happen when you newly encrypt a note?

**Comment by DanteIgashu** [comment]
@pbek (Btw. normally you want to edit or view a encrypted note, not actually decrypt it.)
Sorry, I meant that, edit an encrypted note.
This is the second time that it happens to me, but this second time I don't know why it happened.

The first time that this happened (which is freaking annoying, because I lost my PRECIOUS notes, twice at this point), it happened because my system froze when I had the application opened. I always minimize it to tray, but that time I was actually using it, and even when I had already saved the note and had left the program unused for a few minutes, once my system froze, and I tried to open the note after restarting the system, it didn't ask me for the password and it still allowed me to "edit" the encrypted note, but instead of getting the actual text I just got gibberish.

The second time I don't really know what happened, because I added something to my note, then saved it, then minimized it to tray and a few minutes later I turned off my system, and once I started it again the next day, I opened my note to remember something that I had wrote there, but it didn't even ask me for the password (again) and a moment later I saw that the gibberish was there again.
What I remember doing this time was write what I wanted to write in the file, then I pressed ctrl + s to save it and an instant later I minimized QownNotes to tray, and that makes me wonder if somehow the file got damaged because I did that, even when it had already showed me that the file had been saved, considering that it took it less than a blink to save.

So I guess you could try ending the process while the app is ready to edit the file and maybe it will cause the program to do what I described here.

THANKS! :D

**Comment by pbek** [comment]
I tried to SIG-KILL QOwnNotes while editing an encrypted note but didn't have any issue editing it after starting QOwnNotes again.
The encryption and write process to disk shouldn't take very long...

I don't know if that can be a side-effect from the flatpak version of QOwnNotes (that is community maintained), but I haven't seen a behavior like this before.

**Comment by DanteIgashu** [comment]
@pbek
(Sorry for the late answer, but I was left without internet service for a week thanks to a storm)
After your comment, I uninstalled the flatpak version and installed the version available in the AUR repository from Arch Linux (qownnotes), in the hope that the problem would be just a bug from the flatpak version, but, once I tried to open my note on the AUR version, the same thing happened, it didn't ask me for my password, it just opened the note anyway and showed me some gibberish once more, so I really don't know what is going on.
Also, a couple of days ago the same thing happened on another machine, it was in sleep mode when there was a power outage and once I started my machine again, qownnotes (flatpak) "welcomed" me with the same problem in my notes on that machine, which, to be honest, is pretty freaking annoying, and even when I really love Qownnotes design, I decided to start using another app for my notes, because I don't want to lose my notes every time something "funny" happens with Qownnotes.
So, is there a way that I can try to recover my notes, like by using another program or a website or something?

THANKS in advance and I hope that there's a way to recover my notes. : D
